Extensions/xkit_patches.js

```diff
@@ -915,6 +915,183 @@ XKit.extensions.xkit_patches = new Object({
+				 * @param {String} post_id
+				 * @return {Object} Interface Post Object of post with given id
+				 */
+				find_post: async function(post_id) {
+					// Return a post object based on post ID.
+					var post = $(`[data-id='${post_id}']`);
+
+					if (post.length > 0) {
+						return await XKit.interface.react.post(post);
+					} else {
+						var m_error = {};
+						m_error.error = true;
+						m_error.error_message = "Object not found on page.";
+						return m_error;
+					}
+				},
+
+				control_button_template: null,
+				get_control_button_template: async function() {
+					await XKit.css_map.getCssMap();
+
+					var selector = XKit.css_map.keyToClasses("controlIcon").map(css => `[data-id]:first footer .${css}:first`).join(", ");
+					var control = $(selector);
+
+					var get_used_class_from_map = function(key) {
+						const keyCss = XKit.css_map.keyToCss(key);
+						var element = control.find(keyCss);
+
+						return element.attr("class");
+					};
+
+					var controlIconClass = control.attr("class");
+					var buttonClass = get_used_class_from_map("button");
+
+					var new_control = `
+						<div class="${controlIconClass} {{className}} xkit-interface-control-button" title="{{text}}" {{additional}}>
+							<button class="${buttonClass}" aria-label="" tabindex="0">
+								<div class="xkit-interface-icon" {{data}}></div>
+							</button>
+						</div>
+					`;
+
+					this.control_button_template = new_control;
+
+					return this.control_button_template;
+				},
+				/**
+				 * Create a specification for a control button that can be added to
+				 * future posts using `XKit.interface.add_control_button`.
+				 * @param {String} class_name - CSS class of the button to be created
+				 * @param {String} icon - URL of the button's icon
+				 * @param {String} text - Hover text of the button
+				 * @param {EventListener} func - Function called on click of control button
+				 * @param {String?} ok_icon - URL of icon displayed when the button is
+				 *                            "completed" (e.g. reblog button turning green)
+				 */
+				create_control_button: async function(class_name, icon, text, func, ok_icon) {
+					if (this.control_button_template == null) {
+						this.control_button_template = await this.get_control_button_template();
+					}
+
+					XKit.interface.added_icon.push(class_name);
+					XKit.interface.added_icon_icon.push(icon);
+					XKit.interface.added_icon_text.push(text);
+
+					XKit.tools.add_css(`.${class_name} .xkit-interface-icon {
+						background-image: url('${icon}');
+						background-size: 100% 100%;
+						width: 21px;
+						height: 21px;
+					}`, `xkit_interface_icon__${class_name}`);
+
+					if (typeof ok_icon !== "undefined") {
+						XKit.tools.add_css(`.${class_name} .xkit-interface-icon-completed {
+							background-image: url('${ok_icon}');
+						}`, `xkit_interface_icon__completed__${class_name}`);
+					}
+
+					$(document).on('click', '.' + class_name, function() {
```

Comment:
this will sadly be broken on peepr since new-dash peepr stops propagation of `click` events down to the `document`. this should be changed to somehow automatically apply the callback function directly to each added button, if possible